### PR TITLE
[stable/prometheus-postgres-exporter] add support for Google Cloud SQL

### PR DESCRIPTION
**Pull request by mikeraimondi**
**What this PR does / why we need it**:

Cloud SQL, Google's managed database product, requires use of an authentication
proxy when accessed from dynamic IPs. To limit access to the proxy, it is
recommended to use a sidecar.
More info: https://cloud.google.com/sql/docs/postgres/connect-kubernetes-engine

Cloud SQL, like other managed databases, does not allow use of
superuser accounts. Therefore, add support for a configurable
database in `datasource`.
More info: https://github.com/wrouesnel/postgres_exporter#running-as-non-superuser

### CODE REVIEW

File: stable/prometheus-postgres-exporter/templates/deployment.yaml [hunk 1]
```diff
@@ -55,6 +55,18 @@ spec:
             - name: queries
               mountPath: /etc/config.yaml
               subPath: config.yaml
+    {{- if .Values.cloudSQL.enabled  }}
```
Comment (gianrubio) [inline stable/prometheus-postgres-exporter/templates/deployment.yaml]: I'd rather add an `extraContainer` here so it'll be easier to other users customise it.
Take a look on this [example](https://github.com/kubernetes/charts/blob/24e6370f09a3fbca58f2e761e7fba3fa2e702ec3/stable/elastabot/templates/deployment.yaml#L39)

File: stable/prometheus-postgres-exporter/templates/deployment.yaml [hunk 2]
```diff
@@ -71,4 +83,9 @@ spec:
       - configMap:
           defaultMode: 420
           name: {{ template "prometheus-postgres-exporter.fullname" . }}
-        name: queries
\ No newline at end of file
+        name: queries
+    {{- if .Values.cloudSQL.enabled  }}
```
Comment (gianrubio) [inline stable/prometheus-postgres-exporter/templates/deployment.yaml]: same here, add `extraConfigmap:`

ex:
```
extraConfigmap: 
  - name: cloudsql-instance-credentials
    secret:
      secretName: my-secret-name
```

**Review by gianrubio — CHANGES_REQUESTED** [review]
pleas read my comments

**Comment by unguiculus** [comment]
I agree with @gianrubio. Have a look at the Keycloak chart for reference. The readme documents how to the proxy.

https://github.com/kubernetes/charts/tree/master/stable/keycloak

**Comment by mikeraimondi** [comment]
@gianrubio @unguiculus great feedback, wasn't aware of the `extraContainers` pattern. Made the change, let me know how it looks.

**Pull request merged**